crates/ironrdp-acceptor/src/connection.rs

```diff
@@ -310,7 +311,16 @@ impl Sequence for Acceptor {
             }
 
             AcceptorState::InitiationSendConfirm { requested_protocol } => {
-                let protocol = requested_protocol & self.security;
+                let protocols = requested_protocol & self.security;
+                let protocol = if protocols.intersects(SecurityProtocol::HYBRID_EX) {
+                    SecurityProtocol::HYBRID_EX
+                } else if protocols.intersects(SecurityProtocol::HYBRID) {
+                    SecurityProtocol::HYBRID
+                } else if protocols.intersects(SecurityProtocol::SSL) {
+                    SecurityProtocol::SSL
+                } else {
+                    return Err(ConnectorError::general("Failed to negotiate security protocol"));
```

Comment:
style: https://github.com/Devolutions/IronRDP/blob/3e738a96ed5952553bfa8705c7242d83989a9276/STYLE.md#formatting-of-error-messages
```suggestion
                    return Err(ConnectorError::general("failed to negotiate security protocol"));
```